src/errorpage.cc

```diff
@@ -583,9 +584,6 @@ ErrorDynamicPageInfo::ErrorDynamicPageInfo(const int anId, const char *aName, co
         filename = filenameOrUri;
     }
 
-    const auto info = this; // source code change reduction hack
-    // TODO: Move and refactor to avoid self_destruct()s in reconfigure.
```

Comment:
This is resolved by that -k parse behaviour you keep stating does not exist, but is already widely used. Admin (or automation) can check config changes without halting their running proxy using -k parse.

Reply:
> This is resolved by that -k parse behaviour

Today, `-k parse` does not make Squid reconfiguration errors non-fatal. In many cases, `-k parse` can indeed prevent bad configurations at earlier stages, but this useful tool is not a proper solution to the problem stated in the removed TODO because it is not integrated into reconfiguration to protect the running instance and because it is not compatible with some complex configurations.

I hope we do not need to argue about this. Removal of this (apparently controversial) TODO is outside this PR scope. To avoid creating a stale self_destruct() reference, we can rephrase this TODO as "Refactor to avoid fatal errors during reconfiguration" and leave it be.

> you keep stating does not exist

Let's avoid phrasing that can be easily mistaken for an ad hominem attack. I am not senile enough to keep stating falsehoods.